### DISCUSSION TITLE
Any plans for releasing macOS builds for Apple Silicon?

### DISCUSSION THREAD

**Discussion by marcomasser** [danger/danger-js#1196] [Q&A]
I noticed that the [releases for danger-js](https://github.com/danger/danger-js/releases) contain binaries for macOS (e.g. the [current release 10.8.0](https://github.com/danger/danger-js/releases/download/10.8.0/danger-macos.zip)). This binary is Intel-only (x86_64) and since this is also the one that is used when installing via Homebrew (`brew install danger/tap/danger-js`), users on Apple Silicon Macs (arm64) can’t meaningfully use this for Danger Swift as installed via Homebrew. That’s because Danger Swift does not offer binaries and the code is compiled locally when running `brew install danger/tap/danger-swift`. This results in a situation where the `danger` binary is x86_64, but the `danger-swift` binary is `arm64`.

Honestly, I don’t understand how these two things relate to each other, but running some `danger-swift` commands works (e.g. `danger-swift edit`), but others result in an error (e.g. `danger-swift pr [URL]`):
```
/var/folders/vk/v4gx2v6x4972w9v5r1xggx000000gn/T/_tmp_dangerfile.swift:11:8: error: module 'Danger' was created for incompatible target arm64-apple-macosx10.10: /opt/homebrew/lib/danger/Danger.swiftmodule
import Danger
       ^
```
I guess that this happens:
- Somehow `danger` is run in an x86_64 environment (because it’s compiled only for that)
- `danger` runs `danger-swift` (which inherits the x86_64 environment)
- `danger-swift` tries to load libDanger.dylib – but that doesn’t work because it was only compiled for arm64 💥

I would also guess that this setup would work if the `danger` binary were compatible with arm64 (i.e. either a separate download or a Universal binary). If that is true, are there any plans on releasing binaries for that?

**Comment by orta** [comment]
No plans for me to look into it, but someone else is welcome to
